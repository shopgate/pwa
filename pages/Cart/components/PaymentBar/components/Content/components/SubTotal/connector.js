import { connect } from 'react-redux';
import {
  getCurrency,
  getSubTotal,
} from '@shopgate/pwa-common-commerce/cart/selectors';

/**
 * Maps the contents of the state to the component props.
 * @param {Object} state The current application state.
 * @return {Object} The extended component props.
 */
const mapStateToProps = state => ({
  currency: getCurrency(state),
  value: getSubTotal(state),
});

export default connect(mapStateToProps);
