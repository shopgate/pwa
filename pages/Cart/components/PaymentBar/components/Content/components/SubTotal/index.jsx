import React, { Fragment } from 'react';
import Portal from '@shopgate/pwa-common/components/Portal';
import {
  CART_PAYMENT_BAR_TOTALS_SUB_TOTAL,
  CART_PAYMENT_BAR_TOTALS_SUB_TOTAL_BEFORE,
  CART_PAYMENT_BAR_TOTALS_SUB_TOTAL_AFTER,
} from '@shopgate/pwa-common-commerce/cart/constants/Portals';
import portalProps from '../../totalsPortalProps';
import TotalRow from '../TotalRow';
import Label from './components/Label';
import Amount from './components/Amount';

/**
 * @returns {JSX}
 */
const SubTotal = () => (
  <Fragment>
    <Portal name={CART_PAYMENT_BAR_TOTALS_SUB_TOTAL_BEFORE} props={portalProps} />
    <Portal name={CART_PAYMENT_BAR_TOTALS_SUB_TOTAL} props={portalProps}>
      <TotalRow>
        <Label />
        <Amount />
      </TotalRow>
    </Portal>
    <Portal name={CART_PAYMENT_BAR_TOTALS_SUB_TOTAL_AFTER} props={portalProps} />
  </Fragment>
);

export default SubTotal;
