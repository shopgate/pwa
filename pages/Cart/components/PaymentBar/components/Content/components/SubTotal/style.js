import { css } from 'glamor';
import colors from 'Styles/colors';

const disabled = css({
  color: `${colors.shade4} !important`,
}).toString();

const totalValue = css({
  textAlign: 'right',
}).toString();

export default {
  disabled,
  totalValue,
};
