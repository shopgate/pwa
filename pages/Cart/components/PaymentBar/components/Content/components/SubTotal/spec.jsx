import React from 'react';
import { shallow } from 'enzyme';
import SubTotal from '.';

jest.mock('Pages/Cart/context', () => ({
  Consumer: jest.fn(({ children }) => children({
    currency: 'EUR',
  })),
}));
jest.mock('./connector', () => cmp => cmp);

describe('<SubTotal>', () => {
  it('should render line', () => {
    const wrapper = shallow(<SubTotal amount={10} />).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
