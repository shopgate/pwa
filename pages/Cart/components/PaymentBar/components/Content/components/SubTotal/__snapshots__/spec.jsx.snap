// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubTotal> should render line 1`] = `
<Fragment>
  <Portal
    name="cart.payment-bar.totals.sub-total.before"
    props={null}
  />
  <Portal
    name="cart.payment-bar.totals.sub-total"
    props={null}
  >
    <CartTotalLine
      isDisabled={false}
      type="subTotal"
    >
      <Label
        label="cart.subtotal"
        labelParams={Object {}}
        showSeparator={true}
      />
      <Amount
        amount={10}
        currency="EUR"
      />
    </CartTotalLine>
  </Portal>
  <Portal
    name="cart.payment-bar.totals.sub-total.after"
    props={null}
  />
</Fragment>
`;
